### PR TITLE
IBX-5644 - Email triggers

### PR DESCRIPTION
**Pull request by dabrt**
| Question      | Answer |
| ------------- | --- |
| JIRA Ticket   | [IBX-5644](https://issues.ibexa.co/browse/IBX-5644) |
| Versions      | master |
| Edition       |  |

Add developer-side documentation of the e-mail trigger feature

#### Checklist

- [x] Text renders correctly
- [x] Text has been checked with vale
- [x] Added link to this PR in relevant JIRA ticket or code PR

### CODE REVIEW

File: docs/personalization/integrate_recommendation_service.md [hunk 1]
```diff
@@ -151,6 +151,63 @@ if ($recommendations && isset($recommendations->recommendationResponseList)) {
...
+
+To let your visitors receive emails with recommendations:
+
+1\. With the [User API](api_reference/user_api.md#post-requests), add the `e-mail` attribute to the user record.
+
+2\. Prepare an endpoint that will intercept push messages and pass them on to your mailing system, for example, an Ibexa Connect [webhook](https://doc.ibexa.co/projects/connect/en/latest/tools/webhooks/). The webhook must meet the following requirements:
+
+   - cannot require authentication
+   - must support POST requests
+   - must accept JSON objects in a format that resembles the following example:
```
Comment (kaborkowski) [inline docs/personalization/integrate_recommendation_service.md:170]: Please update JSON example to match prepared template (we decided to simplify reco response): https://issues.ibexa.co/secure/attachment/41005/trigger_message_example.json

File: docs/personalization/integrate_recommendation_service.md [hunk 2]
```diff
@@ -151,6 +151,61 @@ if ($recommendations && isset($recommendations->recommendationResponseList)) {
...
+
+By using [email triggers]([[= user_doc =]]/personalization/triggers.md), your organization can send emails to individual visitors, for example, to invite them to return to the website or remind them of items abandoned in a cart.
+
+Email triggers are push messages with recommendations.
+Specific conditions, such as the time that must pass before email start being sent, content types and attributes to be included in a response, or a number of repetitions, are first defined based on an interview between you and Ibexa.
+Email triggers are then processed on the Personalization server and responses are delivered to a dedicated endpoint.
+
+To let your visitors receive emails with recommendations:
+
+1\. With the [User API](api_reference/user_api.md#post-requests), add the `e-mail` attribute to the user record.
```
Comment (dkorostynski) [inline docs/personalization/integrate_recommendation_service.md:164]: here please add e-mail or **another identifier**

**Review by dkorostynski — CHANGES_REQUESTED** [review]

File: docs/personalization/how_it_works.md [hunk 3]
```diff
@@ -9,7 +9,7 @@ The idea behind the Personalization service is built upon four main steps.
 1. A user visits a news service or an online store, navigates through it and leaves some footprints. The most popular footprints are selecting (CLICK) and purchasing products (BUY).
 2. This information is sent to the Personalization client every time these events happen.
 3. The Personalization server accumulates the tracked events, categorizes them and calculates recommendations for every product and every user in the shop.
-4. Recommendations can be fetched from the Personalization service and presented to users during their sessions.
+4. Recommendations can be fetched from the Personalization service and presented to users during their sessions or pushed by the service to a dedicated endpoint and then sent in an email.
```
Comment (kaborkowski) [inline docs/personalization/how_it_works.md:12]: Should we mark here `then sent in an email`? It is customer responsibility what `dedicated endpoint` will do with those recommendations.

**Review by kaborkowski — APPROVED** [review]

**Review by dkorostynski — APPROVED** [review]

File: docs/personalization/integrate_recommendation_service.md [hunk 4]
```diff
@@ -151,6 +151,62 @@ if ($recommendations && isset($recommendations->recommendationResponseList)) {
 curl_close($curl);
 ```
 
+## Send emails with recommendations
+
+By using [email triggers]([[= user_doc =]]/personalization/triggers.md), your organization can send emails to individual visitors, for example, to invite them to return to the website or remind them of items abandoned in a cart.
+
+Email triggers are push messages with recommendations.
+Specific conditions, such as the time that must pass before email start being sent, content types and attributes to be included in a response, or a number of repetitions, are first defined based on an interview between you and Ibexa.
```
Comment (MagdalenaZuba) [inline docs/personalization/integrate_recommendation_service.md:159]: Specific conditions, such as the time that must pass before email start being sent -> passive voice and weird phrasing, did you mean start of email communication, newsletter chain?
content types -> Content Types
Please update the sentence, maybe a list would help. It is hard to follow and understand.

File: docs/personalization/integrate_recommendation_service.md [hunk 5]
```diff
@@ -151,6 +151,62 @@ if ($recommendations && isset($recommendations->recommendationResponseList)) {
 curl_close($curl);
 ```
 
+## Send emails with recommendations
+
+By using [email triggers]([[= user_doc =]]/personalization/triggers.md), your organization can send emails to individual visitors, for example, to invite them to return to the website or remind them of items abandoned in a cart.
```
Comment (juskora) [inline docs/personalization/integrate_recommendation_service.md:156]: The link is broken. Only visitors? Clients, customers too. I would change the order in the sentence to more active and calling users to action.

File: docs/personalization/integrate_recommendation_service.md (hunk 4 above)
Comment (juskora) [inline docs/personalization/integrate_recommendation_service.md:159]: An interview? Not license agreement and conditions? Sounds weird too. It's a very long and hard to understand sentence.

File: docs/personalization/integrate_recommendation_service.md (hunk 5 above)
Comment (dabrt) [reply docs/personalization/integrate_recommendation_service.md:156]: Agreed re. order, while "client" and "customers" are used in a different meaning throughout perso:
"Set up customer credentials", "PERSONALIZATION_CUSTOMER_ID" refer to Ibexa's customers
"client side", "client-server architecture" refer to Personalization client
"visitor" is used consistently to refer to people who browse and buy
